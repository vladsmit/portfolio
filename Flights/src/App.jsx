import React, { useState, useCallback, useEffect } from 'react';
import PropTypes from 'prop-types';
import connect from 'react-redux/es/connect/connect';
import { bindActionCreators } from 'redux';
import { fetchFlights } from './store/actions/getFlightsFromJson';
import { getFilteredFlights } from './store/actions/getFilteredFlights';
import { spinnerSwitch, addFilters } from './store/actions/flightActions';
import './styles/styles.scss';

import TicketPage from './pages/TicketPage';

const App = ({ fetchFlights, getFilteredFlights, addFilters, filteredFlights, error, loading, filteredAirlines, spinnerSwitch }) => {
  const [sortType, setSortType] = useState(() => {
    return 'rise';
  });

  const [filterType, setFilterType] = useState(() => {
    return [];
  });

  const [priceFrom, setPriceFrom] = useState(() => {
    return null;
  });

  const [priceTo, setPriceTo] = useState(() => {
    return null;
  });

  const [airlines, setAirlines] = useState(() => {
    return [];
  });

  useEffect(() => {
    fetchFlights();
  }, [fetchFlights]);

  const handleGetFlights = useCallback(() => {
    spinnerSwitch(true);
    addFilters({
      sort: sortType,
      filter: filterType,
      priceFrom: priceFrom,
      priceTo: priceTo,
      airlines: airlines
    });
    getFilteredFlights();
    setTimeout(() => {
      spinnerSwitch(false);
    }, 1000);
  }, [spinnerSwitch, addFilters, getFilteredFlights, sortType, filterType, priceFrom, priceTo, airlines]);

  const handleSortTypeStatus = useCallback(e => {
    setSortType(() => {
      return e.target.value;
    });
  }, []);

  const handleFilterTypeStatus = useCallback(checkedValues => {
    setFilterType(() => {
      return checkedValues;
    });
  }, []);

  const handlePriceFromStatus = useCallback(e => {
    setPriceFrom(() => {
      return +e.target.value;
    });
  }, []);

  const handlePriceToStatus = useCallback(e => {
    setPriceTo(() => {
      return +e.target.value;
    });
  }, []);

  const handleAirlinesStatus = useCallback(checkedValues => {
    setAirlines(() => {
      return checkedValues;
    });
  }, []);

  return <TicketPage filteredFlights={filteredFlights} sortType={sortType} loading={loading} error={error} sortTypeStatus={handleSortTypeStatus} filterTypeStatus={handleFilterTypeStatus} priceFromStatus={handlePriceFromStatus} priceToStatus={handlePriceToStatus} airlinesStatus={handleAirlinesStatus} getFlights={handleGetFlights} filteredAirlines={filteredAirlines} />
}

App.propTypes = {
  fetchFlights: PropTypes.func,
  spinnerSwitch: PropTypes.func,
  getFilteredFlights: PropTypes.func,
  addFilters: PropTypes.func,
  error: PropTypes.string,
  loading: PropTypes.bool,
  filteredFlights: PropTypes.array,
  filteredAirlines: PropTypes.array,
}

const mapStateToProps = ({ flightReducer }) => ({
  error: flightReducer.error,
  loading: flightReducer.loading,
  filteredAirlines: flightReducer.filteredAirlines,
  filteredFlights: flightReducer.filteredFlights,
});

const mapDispatchToProps = dispatch => bindActionCreators({ fetchFlights, spinnerSwitch, getFilteredFlights, addFilters }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(App);
