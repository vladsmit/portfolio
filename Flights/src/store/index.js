import { createStore, applyMiddleware, compose } from 'redux';
import { createBrowserHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';
import thunk from 'redux-thunk';

import RootReducers from './reducers';
import middlewares from './middlewares';

export const history = createBrowserHistory();

const composer = window.__REDUX_DEVTOOLS_EXTENSION__ ? compose(applyMiddleware(routerMiddleware(history), thunk, ...middlewares), window.__REDUX_DEVTOOLS_EXTENSION__(),) : compose(applyMiddleware(routerMiddleware(history), thunk, ...middlewares),);

export const store = createStore(RootReducers(history), composer,);