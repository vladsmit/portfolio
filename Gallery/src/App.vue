<template>
    <layout />
</template>

<script>
import { mapActions } from "vuex";
import Layout from "./views/Layout";
export default {
    name: "App",

    components: {
        Layout,
    },

    methods: {
        ...mapActions(["getPhotosFromApi"]),
    },

    created() {
        this.getPhotosFromApi();
    },
};
</script>
