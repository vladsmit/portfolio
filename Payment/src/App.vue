<template>
    <div id="app">
        <ProfileMenu />
        <div class="main--wrapper">
            <router-view />
            <Footer />
        </div>
    </div>
</template>

<script>
import ProfileMenu from "@/components/ProfileMenu.vue";
import Footer from "@/components/Footer.vue";

export default {
    name: "App",
    components: {
        ProfileMenu,
        Footer,
    },
};
</script>

<style lang="scss">
body {
    width: 100vw;
    height: 100vh;
    background-color: black;
    margin: 0 auto;
}

#app {
    width: 700px;
    display: flex;
    margin: 0 auto;
    padding-top: 40px;
}

.main--wrapper {
    width: 67%;
}

@media only screen and (max-width: 480px) {
    #app {
        width: 480px;
        flex-direction: column;
        padding: 0;
    }

    .main--wrapper {
        width: 100%;
    }
}
</style>