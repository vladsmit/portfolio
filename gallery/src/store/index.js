import Vue from 'vue'
import Vuex from 'vuex'
import Axios from 'axios'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    categories: null,
    loading: false
  },

  getters: {
    getLoadingStatus(state) {
      return state.loading;
    },

    getCategories(state) {
      return state.categories;
    }
  },

  mutations: {
    isLoading(state, payload) {
      state.loading = payload;
    },

    savePhotosInState(state, payload) {
      let photos = [];
      let num = 0;
      for (let i = 0; i < 4; i++) {
        photos.push([]);
        for (let j = num; j < 24; j++) {
          if (photos[i].length < 6) {
            photos[i].push(payload[j]);
          } else {
            break;
          }
        }
        num += 6;
      }
      state.categories = photos;
    },
  },

  actions: {
    async getPhotosFromApi({ commit }) {
      commit('isLoading', true);
      const { data } = await Axios.get("https://jsonplaceholder.typicode.com/photos");
      setTimeout(() => {
        commit('savePhotosInState', data);
        commit('isLoading', false);
      }, 500);
    }
  },
})
