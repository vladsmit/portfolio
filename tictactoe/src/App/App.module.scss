.container,
.wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
}

.container {
  width: 100vw;
  min-height: 100vh;
  padding: 20px 0;
}

.wrapper {
  width: 450px;
  flex-direction: column;
}

.gameFieldMargin {
  margin-bottom: 15px;
}
