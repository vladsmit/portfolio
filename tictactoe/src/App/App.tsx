import { useState } from 'react';

import { NewSeriesModal, Header, Footer, GameField } from '../components';
import styles from './App.module.scss';

export const App = () => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <div className={styles.container}>
      <div className={styles.wrapper}>
        <Header setIsOpen={() => setIsOpen(true)} />
        <GameField className={styles.gameFieldMargin} />
        <Footer />
      </div>
      {isOpen && <NewSeriesModal isOpen={isOpen} setIsOpen={setIsOpen} />}
    </div>
  );
};
