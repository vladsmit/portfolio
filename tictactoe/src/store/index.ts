import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import ticTacToeReducer from './slices/ticTacToeSlice';

export const store = configureStore({
  reducer: {
    ticTacToe: ticTacToeReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
