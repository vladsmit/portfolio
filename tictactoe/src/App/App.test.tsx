import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import { store } from '../store';
import { App } from './App';

describe('App component', () => {
  it('should render App component', () => {
    const component = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });
});
