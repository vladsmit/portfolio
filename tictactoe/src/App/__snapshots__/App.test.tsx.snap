// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="wrapper"
        >
          <h1
            class="text"
          >
            Крестики-нолики
          </h1>
          <div
            class="gameCountBlock"
          >
            <div
              class="playerInfoBlock"
            >
              <span
                class="playerName"
              >
                Игрок 1
              </span>
              <span
                class="text"
              >
                0
              </span>
            </div>
            <button
              class="ant-btn css-dev-only-do-not-override-k7429z ant-btn-text buttonFontWeight"
              type="button"
            >
              <span>
                Новая серия
              </span>
            </button>
            <div
              class="playerInfoBlock"
            >
              <span
                class="playerName"
              >
                Игрок 2
              </span>
              <span
                class="text"
              >
                0
              </span>
            </div>
          </div>
          <div
            class="wrapper gameFieldMargin"
          >
            <div
              class="row"
            >
              <div
                class="cell"
                data-testid="wrapper"
              />
              <div
                class="cell"
                data-testid="wrapper"
              />
              <div
                class="cell"
                data-testid="wrapper"
              />
            </div>
            <div
              class="row"
            >
              <div
                class="cell"
                data-testid="wrapper"
              />
              <div
                class="cell"
                data-testid="wrapper"
              />
              <div
                class="cell"
                data-testid="wrapper"
              />
            </div>
            <div
              class="row"
            >
              <div
                class="cell"
                data-testid="wrapper"
              />
              <div
                class="cell"
                data-testid="wrapper"
              />
              <div
                class="cell"
                data-testid="wrapper"
              />
            </div>
          </div>
          <div
            class="wrapper"
          >
            <p
              class="text"
            >
              <span>
                Ходит: 
              </span>
              <span
                class="playerText"
              >
                Игрок 1
              </span>
            </p>
            <button
              class="ant-btn css-dev-only-do-not-override-k7429z ant-btn-primary ant-btn-background-ghost"
              type="button"
            >
              <span>
                Новая игра
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="wrapper"
      >
        <h1
          class="text"
        >
          Крестики-нолики
        </h1>
        <div
          class="gameCountBlock"
        >
          <div
            class="playerInfoBlock"
          >
            <span
              class="playerName"
            >
              Игрок 1
            </span>
            <span
              class="text"
            >
              0
            </span>
          </div>
          <button
            class="ant-btn css-dev-only-do-not-override-k7429z ant-btn-text buttonFontWeight"
            type="button"
          >
            <span>
              Новая серия
            </span>
          </button>
          <div
            class="playerInfoBlock"
          >
            <span
              class="playerName"
            >
              Игрок 2
            </span>
            <span
              class="text"
            >
              0
            </span>
          </div>
        </div>
        <div
          class="wrapper gameFieldMargin"
        >
          <div
            class="row"
          >
            <div
              class="cell"
              data-testid="wrapper"
            />
            <div
              class="cell"
              data-testid="wrapper"
            />
            <div
              class="cell"
              data-testid="wrapper"
            />
          </div>
          <div
            class="row"
          >
            <div
              class="cell"
              data-testid="wrapper"
            />
            <div
              class="cell"
              data-testid="wrapper"
            />
            <div
              class="cell"
              data-testid="wrapper"
            />
          </div>
          <div
            class="row"
          >
            <div
              class="cell"
              data-testid="wrapper"
            />
            <div
              class="cell"
              data-testid="wrapper"
            />
            <div
              class="cell"
              data-testid="wrapper"
            />
          </div>
        </div>
        <div
          class="wrapper"
        >
          <p
            class="text"
          >
            <span>
              Ходит: 
            </span>
            <span
              class="playerText"
            >
              Игрок 1
            </span>
          </p>
          <button
            class="ant-btn css-dev-only-do-not-override-k7429z ant-btn-primary ant-btn-background-ghost"
            type="button"
          >
            <span>
              Новая игра
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
